cmd/surf/services/parallelextraction.go

```diff
@@ -0,0 +1,69 @@
+package services
+
+import (
+	"context"
+	"github.com/CloudHub360/ch360.go/ch360"
+	"github.com/CloudHub360/ch360.go/ch360/results"
+	"github.com/CloudHub360/ch360.go/pool"
+	"github.com/pkg/errors"
+	"io"
+	"os"
+)
+
+//go:generate mockery -name "FileExtractor"
+
+type FileExtractor interface {
+	Extract(ctx context.Context, fileContent io.Reader, extractorName string) (*results.ExtractionResult, error)
+}
+
+// ParallelExtractionService wraps the ch360.FileExtractor to process multiple files in parallel.
+type ParallelExtractionService struct {
```

Comment:
💪 It would be nice to have a single "Run this struct/func in parallel" interface rather than duplicating it for classification, extraction and x, y, z other things. It would also mean only having to test the parallelisation once.

Reply:
That's pretty much what the `pool.Pool` struct does, which this uses under the bonnet.